packages/react-native-nitro-avalabs-crypto/src/Crypto.ts

```diff
@@ -0,0 +1,364 @@
+  con.log('[Crypto] pointAddScalar called')
+  const pkAB = hexLikeToArrayBuffer(publicKey as unknown as string)
+  const twAB = hexLikeToArrayBuffer(tweak as unknown as string)
+  const out = NativeCrypto.pointAddScalar(pkAB, twAB, isCompressed)
+  return new Uint8Array(out)
+}
+
+/** ECDSA sign. Message must be a 32-byte digest. Returns DER signature bytes. */
+function isDerSignature(u8: Uint8Array): boolean {
+  return u8.length >= 8 && u8[0] === 0x30
+}
+
+/** Convert 32-byte big-endian integer to minimal DER INTEGER (with zero prefix if high bit set). */
+function be32ToDerInt(src: Uint8Array): Uint8Array {
+  // Strip leading zeros
+  let i = 0
+  while (i < src.length - 1 && src[i] === 0) i++
+  let v = src.subarray(i)
+  // If high bit set, prepend 0x00 to force positive
+  if (v[0] & 0x80) {
+    const out = new Uint8Array(v.length + 1)
+    out[0] = 0x00
+    out.set(v, 1)
+    return out
+  }
+  return v
+}
+
+/** Convert a 64-byte compact ECDSA signature (r||s) to DER sequence. */
+function compact64ToDer(sig64: Uint8Array): Uint8Array {
+  if (sig64.length !== 64)
+    throw new TypeError('compact ECDSA signature must be 64 bytes')
+  const r = sig64.subarray(0, 32)
+  const s = sig64.subarray(32, 64)
+  const rDer = be32ToDerInt(r)
+  const sDer = be32ToDerInt(s)
+  const len = 2 + rDer.length + 2 + sDer.length // 0x02 rLen r 0x02 sLen s
+  const out = new Uint8Array(2 + 2 + rDer.length + 2 + sDer.length)
+  let p = 0
+  out[p++] = 0x30 // SEQUENCE
+  out[p++] = len
+  out[p++] = 0x02 // INTEGER
+  out[p++] = rDer.length
+  out.set(rDer, p)
+  p += rDer.length
+  out[p++] = 0x02 // INTEGER
+  out[p++] = sDer.length
+  out.set(sDer, p)
+  return out
+}
+
+export function sign(
+  secretKey: string | ArrayBuffer | Uint8Array,
+  message: string | ArrayBuffer | Uint8Array
+): Uint8Array {
+  con.log('[Crypto] sign called')
+  const skAB = hexLikeToArrayBuffer(secretKey as any)
+  const msgAB = hexLikeToArrayBuffer(message as any)
+  if (msgAB.byteLength !== 32)
+    throw new TypeError('ECDSA message must be 32 bytes')
+  const out = NativeCrypto.sign(skAB, msgAB)
+  let sig = new Uint8Array(out)
+  // Normalize to compact(64) – many JS libs expect r||s
+  if (isDerSignature(sig)) {
+    con.log('[Crypto] sign: converting DER signature to compact(64)')
+    sig = derToCompact64(sig)
+  } else if (sig.length !== 64) {
+    throw new TypeError(`ECDSA signature has unexpected length: ${sig.length}`)
+  }
+  return sig
+}
+
+/** Parse minimal ASN.1 DER ECDSA signature and return {r,s} raw big-endian bytes (unpadded). */
+function parseDerEcdsa(sig: Uint8Array): { r: Uint8Array; s: Uint8Array } {
+  if (sig.length < 8 || sig[0] !== 0x30)
+    throw new TypeError('Invalid DER: no SEQ')
+  let p = 1
+  let len = sig[p++]
+  if (len & 0x80) {
+    const n = len & 0x7f
+    if (n === 0 || n > 2) throw new TypeError('Invalid DER: long len too big')
+    if (p + n > sig.length) throw new TypeError('Invalid DER: length overflow')
```

Comment:
can you add tests for these functions?

Reply:
Tests for crypto functioned were implemented  separately in a testecc.js script and will be used in a separate PR